airflow/executors/kubernetes_executor.py

```diff
@@ -576,6 +576,7 @@ def _labels_to_key(self, labels):
     def terminate(self):
         """Termninates the watcher."""
         self.watcher_queue.join()
+        self.kube_watcher.join()
```

Comment:
I don't think this is right. This doesn't signal in any way to the kube_watcher that it should shutdown - it simply blocks execution until it stops.

Reply:
Correct me if I am wrong @ashb, I believe this call [self.processor_agent.terminate()](https://github.com/apache/airflow/blob/master/airflow/jobs/scheduler_job.py#L1434) signals the `manager` to terminate.

Reply:
No not at all. This has nothing to do with the scheduler_job

Reply:
Thanks @ashb. I have added `terminate` call.